### DISCUSSION TITLE
Product Roadmap

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#218] [General]
Here is my high-level view of the feature roadmap for this library.  You can also see a more tactical view in the working [Backlog](https://github.com/DaveSkender/Stock.Indicators/projects/1).  Feedback appreciated => visit topic links below or add new ideas to the discussion board.  In addition to features, we'll also opportunistically add new indicators throughout all versions.

## Version 1.x - baseline time-series indicators

- [x] baseline set of indicators
- [x] more standard indicators in the current "time-series" use-case format
- [x] general updates to related documentation
- [x] #656 

## Version 2.x - indicator chaining

- [x] #802
    - enables #630
- [ ] #220 (e.g. cup and handle, hanging man, shooting star, etc.)
- [x] small things that can’t be fixed in v1
  - [x] reverse order of `quotes` and `market` list parameters so extension methods are consistent (Beta, PRS, etc.)
  - [x] remove v1 `Obsolete` backwards compatibility accommodations

## Version 3.x - incremental streaming use cases

- [ ] #1014 (one-bar-at-a-time calculations), see #1018

## Version 4.x - intrinsic analysis

- [ ] #320 (intrinsic only), a broader signal generator or backtest engine would be in a separate project
    - see also #822 for examples of intrinsic signals

**Comment by bobtabor** [comment]
First of all, like I posted in an issue I reported yesterday, this library is amazing.  I absolutely love it.  I'm using this library to generate buy and sell signals in an algorithmic trading system I'm building for myself (non-commercial).  I was resigned to paying $20 - $50/month for an API that does this with virtually non-existent docs.

My goal is to replicate TradingView's "Technicals" scorecard, which combines a subset of all the technical indicators you supply in your library and generates a simple "buy", "sell" or "neutral" recommendation.  (See screenshots in case you're not familiar with this.)

For my own selfish purposes, I might request one of two things:

1. A simplified API surface where I merely supply the data from Polygon and you kick back the scorecard.  Yes, I'm asking you to do all the hard work I am doing.  :wink:   Besides laziness, I'm concerned that I'm not interpreting the values from your methods correctly.  

There are several examples I could point to, but a great example is `GetElderRay()` ... [Investopedia](https://www.investopedia.com/terms/e/elderray.asp) says the following about interpreting its results:

> Technical traders should consider long positions if the bull power is rising, bear power is in negative territory and rising (getting weaker), and EMA is sloping upward.

The challenge here is that I need to calculate the *slope* / *trend* of both the bull power and the EMA.  I don't know how to do that.  I mean, I guess I could use some sort of linear regression algorithm to determine slope of the data, but how many periods should I use?  I guess I just don't know this space well enough to make good decisions about how to utilize the results from your library.  All I'm looking for is a "buy", "neutral", or "sell".  I'm stitching together recommendations from around the web (Wikipedia, Investopedia, etc.) on how the results are typically interpreted and that leaves much to be desired.

2.  In lieu of the first request, perhaps you would be willing to implement a simple slope / trend utility method (maybe that's already there?) that forces me to give you the correct amount of data to determine the direction of the market.  I guess I could implement it myself, but I would always be wondering if I'm doing the right thing.  Again, I just don't trust my inexperience with technical indicators enough yet to have confidence in what I'm doing here.

BTW, candlestick patterns would be awesome!

Thanks again!




![sidebar](https://user-images.githubusercontent.com/45040176/129210839-6ed87e27-dbf7-473a-b2c1-764c2ddc5fda.PNG)


![top](https://user-images.githubusercontent.com/45040176/129210849-e00b22ed-d234-499e-ba1f-a2ba7a6590e0.PNG)


![bottom](https://user-images.githubusercontent.com/45040176/129210862-9090ddc6-d220-4e36-b8e1-0139958ff81c.PNG)

**Reply by DaveSkender** [reply]
Thanks for sharing Bob, this is a really cool use case.  I've seen this sort of aggregation done with CNN's [Fear and Greed Index](https://money.cnn.com/data/fear-and-greed).  And I agree, paying a lot for an external REST API to get undocumented indicators that are super slow over HTTP seems like a bad deal, especially for personal use and since all of these recipes are in the public domain.

Generally speaking, I like your suggestions here; however, I'd probably implement in a separate project as this library is intentionally limited to: quotes in, indicator data out -- I'm trying to stick to the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).  In fact, this library was started as part of my own private algo trading platform, but then I decided to refactor for separation of concerns.

For your specific recommendations:

> A simplified API surface where I merely supply the data from Polygon and you kick back the scorecard. Yes, I'm asking you to do all the hard work I am doing. 😉 Besides laziness, I'm concerned that I'm not interpreting the values from your methods correctly.

I had not thought of a Scorecard per se, but I am thinking about a separate project for backtesting that can generate signals (see #484 for discussion and ideation).  Part of the challenge in both of these cases is how to allow for subjectivity in determining what is "bullish" exactly; for example, [RSI](https://daveskender.github.io/Stock.Indicators/indicators/Rsi/#content) has this general idea that values over 70 and under 30 are bullish/bearish, but some might want 80/20 and some might say those only represent overbought/oversold.  Lots to think about.  In clear cases, we might consider adding extended analysis in this library, but that would be limited to individual indicator data (see #320 for discussion on extended analysis).

> In lieu of the first request, perhaps you would be willing to implement a simple slope / trend utility method (maybe that's already there?) that forces me to give you the correct amount of data to determine the direction of the market. I guess I could implement it myself, but I would always be wondering if I'm doing the right thing. Again, I just don't trust my inexperience with technical indicators enough yet to have confidence in what I'm doing here.

Checkout [Slope and Linear Regression](https://daveskender.github.io/Stock.Indicators/indicators/Slope/#content) and [How to Generate an Indicator of Indicators](https://daveskender.github.io/Stock.Indicators/guide/#generating-indicator-of-indicators).
